packages/features/ee/organizations/pages/settings/attributes/AttributesForm.tsx

```diff
@@ -0,0 +1,110 @@
+import { Button, Form, SelectField, InputField, Label, Input } from "@calcom/ui";
+
+const AttributeSchema = z.object({
+  attrName: z.string().min(1),
+  type: z.enum(["TEXT", "NUMBER", "SINGLE_SELECT", "MULTI_SELECT"]),
+  options: z.array(z.object({ value: z.string(), id: z.string().optional() })),
+});
+
+type FormValues = z.infer<typeof AttributeSchema>;
+
+const AttributeTypeOptions = [
+  { value: "TEXT", label: "Text" },
+  { value: "NUMBER", label: "Number" },
+  { value: "SINGLE_SELECT", label: "Single Select" },
+  { value: "MULTI_SELECT", label: "Multi Select" },
+];
+
+interface AttributeFormProps {
+  initialValues?: FormValues;
+  onSubmit: (values: FormValues) => void;
+  header: React.ReactNode;
+}
+
+export function AttributeForm({ initialValues, onSubmit, header }: AttributeFormProps) {
+  const { t } = useLocale();
+  const [listRef] = useAutoAnimate<HTMLDivElement>({
+    duration: 300,
+    easing: "ease-in-out",
+  });
+
+  const form = useForm<FormValues>({
+    resolver: zodResolver(AttributeSchema),
+    defaultValues: initialValues || {
+      attrName: "",
+      options: [{ value: "" }],
+      type: "TEXT",
+    },
+  });
+
+  const { fields, append, remove } = useFieldArray({
+    control: form.control,
+    name: "options",
+  });
+
+  const watchedType = form.watch("type");
+
+  return (
+    <Form
+      form={form}
+      className="mt-6 flex flex-col space-y-4 lg:space-y-6"
+      handleSubmit={(values) => {
+        onSubmit({
+          ...values,
+          options: values.options,
+        });
+      }}>
+      {header}
+      <InputField label={t("name")} required {...form.register("attrName")} />
+      <Controller
+        name="type"
+        render={({ field: { value, onChange } }) => (
+          <SelectField
+            label="Type"
+            options={AttributeTypeOptions}
+            onChange={(option) => {
+              if (!option) return;
+              onChange(option.value);
+            }}
+            value={AttributeTypeOptions.find((attr) => attr.value === value)}
+            required
+          />
+        )}
+      />
+      {["SINGLE_SELECT", "MULTI_SELECT"].includes(watchedType) && (
+        <div className="bg-muted border-muted mt-6 rounded-lg border p-6">
+          <div className="flex flex-col gap-2">
+            <Label>{t("options")}</Label>
+            <div ref={listRef}>
+              {fields.map((field, index) => (
+                <div className="flex items-center gap-2" key={field.id}>
+                  <Input {...form.register(`options.${index}.value`)} className="w-full" />
+                  <Input {...form.register(`options.${index}.id`)} className="hidden" />
+                  <Button
+                    type="button"
+                    variant="icon"
+                    StartIcon="x"
+                    color="minimal"
+                    className="mb-2"
+                    disabled={index === 0}
```

Comment:
Why do we want to disable deleting the first option? If I have multiple options, what if I wanted to delete the first option?

Should we instead disable it if it's the only option?